Utilities/ocvalidate/ValidateMisc.c

```diff
@@ -0,0 +1,262 @@
+**/
+
+#include "ocvalidate.h"
+#include "OcValidateLib.h"
+
+#include <Library/OcBootManagementLib.h>
+#include <Library/OcConfigurationLib.h>
+#include <Protocol/OcLog.h>
+
+/**
+  Validate if SecureBootModel has allowed value.
+
+  @param[in]  SecureBootModel  SecureBootModel retrieved from user config.
+
+  @retval     TRUE             If SecureBootModel is valid.
+**/
+STATIC
+BOOLEAN
+ValidateSecureBootModel (
+  IN  CONST CHAR8  *SecureBootModel
+  )
+{
+  UINT32   Index;
+  CONST CHAR8 *AllowedSecureBootModel[] = {
+    "Default", "Disabled",
+    "j137",  "j680",  "j132",  "j174",  "j140k",
+    "j780",  "j213",  "j140a", "j152f", "j160",
+    "j230k", "j214k", "j223",  "j215",  "j185", "j185f",
+    "x86legacy"
+  };
+
+  for (Index = 0; Index < ARRAY_SIZE (AllowedSecureBootModel); ++Index) {
+    if (AsciiStrCmp (SecureBootModel, AllowedSecureBootModel[Index]) == 0) {
+      return TRUE;
+    }
+  }
+
+  return FALSE;
+}
+
+UINT32
+CheckMisc (
+  IN  OC_GLOBAL_CONFIG  *Config
+  )
+{
+  UINT32          ErrorCount;
+  UINT32          Index;
+  OC_MISC_CONFIG  *UserMisc;
+  OC_UEFI_CONFIG  *UserUefi;
+  UINT32          ConsoleAttributes;
+  CONST CHAR8     *HibernateMode;
+  UINT32          PickerAttributes;
+  CONST CHAR8     *PickerMode;
+  UINT64          DisplayLevel;
+  UINT64          AllowedDisplayLevel;
+  UINT64          HaltLevel;
+  UINT64          AllowedHaltLevel;
+  UINT32          Target;
+  CONST CHAR8     *BootProtect;
+  BOOLEAN         IsRequestBootVarRoutingEnabled;
+  CONST CHAR8     *AsciiDmgLoading;
+  UINT32          ExposeSensitiveData;
+  CONST CHAR8     *AsciiVault;
+  UINT32          ScanPolicy;
+  UINT32          AllowedScanPolicy;
+  CONST CHAR8     *SecureBootModel;
+  CONST CHAR8     *Arguments;
+  CONST CHAR8     *Comment;
+  CONST CHAR8     *AsciiName;
+  CONST CHAR16    *UnicodeName;
+  CONST CHAR8     *Path;
+
+  DEBUG ((DEBUG_VERBOSE, "config loaded into Misc checker!\n"));
+
+  ErrorCount                     = 0;
+  UserMisc                       = &Config->Misc;
+  UserUefi                       = &Config->Uefi;
+  ConsoleAttributes              = UserMisc->Boot.ConsoleAttributes;
+  HibernateMode                  = OC_BLOB_GET (&UserMisc->Boot.HibernateMode);
+  PickerAttributes               = UserMisc->Boot.PickerAttributes;
+  PickerMode                     = OC_BLOB_GET (&UserMisc->Boot.PickerMode);
+  DisplayLevel                   = UserMisc->Debug.DisplayLevel;
+  AllowedDisplayLevel            = DEBUG_WARN | DEBUG_INFO | DEBUG_VERBOSE | DEBUG_ERROR;
+  HaltLevel                      = DisplayLevel;
+  AllowedHaltLevel               = AllowedDisplayLevel;
+  Target                         = UserMisc->Debug.Target;
+  BootProtect                    = OC_BLOB_GET (&UserMisc->Security.BootProtect);
+  IsRequestBootVarRoutingEnabled = UserUefi->Quirks.RequestBootVarRouting;
+  AsciiDmgLoading                = OC_BLOB_GET (&UserMisc->Boot.PickerMode);
```

Comment:
This is a mistake, should be DmgLoading.

Reply:
https://github.com/acidanthera/OpenCorePkg/pull/169/commits/08bdbce451c29b362124f2a71603740a86beb343